pkg/csi/service/wcpguest/controller.go

```diff
@@ -531,6 +575,75 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 	return &csi.ControllerUnpublishVolumeResponse{}, nil
 }
 
+// controllerUnpublishForFileVolume is helper method to handle ControllerPublishVolume for File volumes
+func controllerUnpublishForFileVolume(ctx context.Context, req *csi.ControllerUnpublishVolumeRequest, c *controller) (
+	*csi.ControllerUnpublishVolumeResponse, error) {
+	log := logger.GetLogger(ctx)
+	// Adding watch on the CnsFileAccessConfig instance to register for updates
+	cnsFileAccessConfigWatcher, err := k8s.NewCnsFileAccessConfigWatcher(ctx, c.restClientConfig, c.supervisorNamespace)
+	if err != nil {
+		msg := fmt.Sprintf("failed to create cnsFileAccessConfigWatcher. Error: %+v", err)
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	// Attacher timeout, default is set to 4 minutes
+	timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
+	cnsFileAccessConfigInstanceName := req.NodeId + "-" + req.VolumeId
+	watchCnsFileAccessConfig, err := cnsFileAccessConfigWatcher.Watch(metav1.ListOptions{
+		FieldSelector:  fields.SelectorFromSet(fields.Set{"metadata.name": cnsFileAccessConfigInstanceName}).String(),
+		TimeoutSeconds: &timeoutSeconds,
```

Comment:
If you perform a get on the CR(before detach) you want to delete and then watch since the CR's revision, the code will deal with less events.

Reply:
Done